microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/ElytronSSLContextRegistry.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.microprofile.reactive.messaging.config.kafka.ssl.context;
+
+import javax.net.ssl.SSLContext;
+
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceRegistry;
+
+/**
+ * @author <a href="mailto:kabir.khan@jboss.com">Kabir Khan</a>
+ */
+public class ElytronSSLContextRegistry {
+
+    private static final ElytronSSLContextRegistry INSTANCE = new ElytronSSLContextRegistry();
+    private static final ServiceName BASE_CLIENT_SSL_CONTEXT_NAME = ServiceName.of("org", "wildfly", "security", "ssl-context");
+
+    private volatile ServiceRegistry serviceRegistry;
+
+    private ElytronSSLContextRegistry() {
+    }
+
+    public static void setServiceRegistry(ServiceRegistry serviceRegistry) {
+        serviceRegistry = serviceRegistry;
+    }
+
+    static SSLContext getInstalledSSLContext(String name) {
```

Comment:
I still think this needs to be split into two methods.

https://github.com/wildfly/wildfly/pull/14475/files#diff-d9d0091e8a499da58a7b0ae2d67bc83b134bb6aaa06beae3fe69404b1ee3898eR86 does not need the SSLContext; it just needs a true | false indicating if the ServiceController exists.

If it does not exist, L51 below will NPE.

If it does exist, but is not started (which could happen as there's no clear dep on the service from the DUPS running ReactiveMessagingSslConfigProcessor) then MSC is allowed to throw an exception at L51, which would unnecessarily fail the deployment.

Reply:
I totally forgot about this point - sorry!